src/ethereumjs-abi-utils.ts

```diff
@@ -82,18 +88,20 @@ function parseTypeN(type) {
  * @param arg - Number to parse.
  * @returns Parsed value.
  */
-export function parseNumber(arg) {
+export function parseNumber(arg: string | number | BN): BN {
   const type = typeof arg;
   if (type === 'string') {
```

Comment:
If you do `typeof arg` here, you can remove the type cast further on.
```suggestion
  if (typeof arg === 'string') {
```

Reply:
Fair. This whole file will be removed in #273 regardless. 😅